priv/user_linux.go

```diff
@@ -100,11 +100,20 @@ func recursiveEnsureOwner(path string, uid, gid int) error {
 	return nil
 }
 
-func RunAs(uid, gid int) error {
+// RunAs sets the user ID and group ID of the calling process.
+// If withUserLookup is true, the user ID must be valid in the image where the process is running.
+func RunAs(uid, gid int, withUserLookup bool) error {
 	if uid == os.Getuid() && gid == os.Getgid() {
 		return nil
 	}
-	user, err := user.LookupId(strconv.Itoa(uid))
+
+	var (
+		foundUser *user.User
+		err       error
+	)
+	if withUserLookup {
+		foundUser, err = user.LookupId(strconv.Itoa(uid))
+	}
```

Comment:
This code could be made simpler if moved below line 134, but I am not sure if there is value in doing the user check before dropping privileges.

Reply:
`/etc/passwd` should be world readable so doing the user check after dropping privs shouldn't cause any problems